src/util/rocksdb/properties.rs

```diff
@@ -266,7 +267,15 @@ impl IndexHandles {
                 v.offset
             }
         };
-        assert!(end_offset >= start_offset);
+        if end_offset < start_offset {
```

Comment:
If we have guidelines of when to `panic!` and  when to `assert!`?



```rust
assert!(end_offset >= start_offset, "start {:?} end {:?} start_offset {} end_offset {}",
                start,
                end,
                start_offset,
                end_offset
            );
```

Reply:
We prefer explicit `panic` in general sources, and `assert` in tests.

Reply:
Oh, I don't know `assert!` can be used like that, I prefer to use `assert!` here.